src/main/java/uk/co/ramp/covid/simulation/population/Population.java

```diff
@@ -224,50 +222,28 @@ public void createMixing() {
         int nConstructionSites = populationSize / PopulationParameters.get().getConstructionSiteRatio();
         int nNurseries = populationSize / PopulationParameters.get().getNurseriesRatio();
         int nRestaurants = populationSize / PopulationParameters.get().getRestaurantRatio();
-        int nEstablishments = nHospitals + nSchools + nShops + nOffices + nConstructionSites + nNurseries + nRestaurants;
-        this.shopIndexes = new int[nShops];
-        this.restaurantIndexes = new int[nRestaurants];
-
-        LOGGER.info("Total number of establishments = {}", nEstablishments);
-
-        CommunalPlace[] places = new CommunalPlace[nEstablishments];
-        for (int i = 0; i < nEstablishments; i++) {
-            if (i < nHospitals) places[i] = new Hospital(i);
-            else if (i < nHospitals + nSchools) places[i] = new School(i);
-            else if (i < nHospitals + nSchools + nShops) { // Allocate shops and their indexes
-                places[i] = new Shop(i);
-                this.shopIndexes[i - nHospitals - nSchools] = i;
-            } else if (i < nHospitals + nSchools + nShops + nOffices) places[i] = new Office(i);
-            else if (i < nHospitals + nSchools + nShops + nOffices + nConstructionSites)
-                places[i] = new ConstructionSite(i);
-            else if (i < nHospitals + nSchools + nShops + nOffices + nConstructionSites + nNurseries)
-                places[i] = new Nursery(i);
-            else if (i < nHospitals + nSchools + nShops + nOffices + nConstructionSites + nNurseries + nRestaurants) {
-                places[i] = new Restaurant(i);
-                this.restaurantIndexes[i - nHospitals - nSchools - nShops - nOffices - nConstructionSites - nNurseries] = i;
-            }
 
-        }
-        this.cPlaces = places;
+        places.createNHospitals(nHospitals);
```

Comment:
For safety, should we set a minimum of 1 of each place?

Reply:
I'm not sure - If we were modelling small places (say the Western Isles) It could be possible there's no construction going on, or public transport etc?